### ISSUE TITLE
Add "arbitrary-precision decimal" Presto type

### ISSUE THREAD

**Issue by findepi** [prestosql/presto#2274]
Some storage systems Presto connects to have a way store a decimal of arbitrary precision.
In Java, this would correspond to `java.math.BigDecimal`.

Examples:

* PostgreSQL's `NUMERIC` (today it's not mapped: https://github.com/prestosql/presto/issues/2079)
* Cassandra's `DECIMAL` (today it's mapped to `double`)

Today, we don't have a really good way of mapping these types in Presto.

**Comment by findepi** [comment]
Of course, such a type would have drawbacks that go beyond implementation and maintenance cost.
E.g. it will always perform worse than currently implemented "long decimal" (128bit), so it's worth considering whether adding such a type wouldn't be like an opening of a Pandora's box.

**Comment by walter-weinmann** [comment]
I'm still very interested in the final solution

**Comment by findepi** [comment]
This might be related to SQL spec's `DECFLOAT` type.

**Comment by martint** [comment]
> This might be related to SQL spec's DECFLOAT type.

No, `DECFLOAT(p)` is sort of a combination of DECIMAL with a traditional floating point type. There's a mantissa with a defined number of digits given by `p` and an exponent. DOUBLE/REAL, on the other hand, define a precision in terms of binary digits.

**Comment by tooptoop4** [comment]
currently it seems impossible to read oracle number with precision 39 or 40

**Comment by findepi** [comment]
cc @jkylling

**Comment by findepi** [comment]
I have drafted a potential implementation for such a type:
- https://github.com/trinodb/trino/pull/28120
